components/authorization-keycloak/backend/src/main/kotlin/migration/RolesToDbMigration.kt

```diff
@@ -0,0 +1,244 @@
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.components.authorization.keycloak.migration
+
+import org.eclipse.apoapsis.ortserver.clients.keycloak.GroupName
+import org.eclipse.apoapsis.ortserver.clients.keycloak.KeycloakClient
+import org.eclipse.apoapsis.ortserver.components.authorization.db.RoleAssignmentsTable
+import org.eclipse.apoapsis.ortserver.components.authorization.keycloak.roles.OrganizationRole
+import org.eclipse.apoapsis.ortserver.components.authorization.keycloak.roles.ProductRole
+import org.eclipse.apoapsis.ortserver.components.authorization.keycloak.roles.RepositoryRole
+import org.eclipse.apoapsis.ortserver.components.authorization.keycloak.roles.Role
+import org.eclipse.apoapsis.ortserver.components.authorization.keycloak.roles.Superuser
+import org.eclipse.apoapsis.ortserver.components.authorization.rights.OrganizationRole as DbOrganizationRole
+import org.eclipse.apoapsis.ortserver.components.authorization.rights.ProductRole as DbProductRole
+import org.eclipse.apoapsis.ortserver.components.authorization.rights.RepositoryRole as DbRepositoryRole
+import org.eclipse.apoapsis.ortserver.components.authorization.rights.Role as DbRole
+import org.eclipse.apoapsis.ortserver.components.authorization.service.AuthorizationService
+import org.eclipse.apoapsis.ortserver.dao.dbQuery
+import org.eclipse.apoapsis.ortserver.dao.repositories.organization.OrganizationsTable
+import org.eclipse.apoapsis.ortserver.dao.repositories.product.ProductsTable
+import org.eclipse.apoapsis.ortserver.dao.repositories.repository.RepositoriesTable
+import org.eclipse.apoapsis.ortserver.model.CompoundHierarchyId
+import org.eclipse.apoapsis.ortserver.model.HierarchyId
+import org.eclipse.apoapsis.ortserver.model.OrganizationId
+import org.eclipse.apoapsis.ortserver.model.ProductId
+import org.eclipse.apoapsis.ortserver.model.RepositoryId
+
+import org.jetbrains.exposed.sql.Database
+
+import org.slf4j.LoggerFactory
+
+private val logger = LoggerFactory.getLogger(RolesToDbMigration::class.java)
+
+/**
+ * A class implementing a migration that moves roles management to the database.
+ *
+ * The migration iterates over the existing organizations, products, and repositories and the associating groups in
+ * Keycloak that represent the user roles on these entities. It then creates corresponding role assignment entries in
+ * the database. The migration needs to be executed once to set up the data structures for the new authorization
+ * component.
+ */
+class RolesToDbMigration(
```

Comment:
IIUC this migration can only run once and user's won't be able to perform a rollback.
You might consider marking this commit as a breaking change and maybe also pointing this out in the commit message.